cmd/kaniko-docker/main.go

```diff
@@ -201,7 +201,7 @@ func run(c *cli.Context) error {
 			ExpandTag:       c.Bool("expand-tag"),
 			Args:            c.StringSlice("args"),
 			Target:          c.String("target"),
-			Repo:            c.String("repo"),
+			Repo:            buildRepo(c.String("registry"), c.String("repo")),
```

Comment:
In buildRepo method, L265 assumes that registry url does not end with /. In case registry url already contains "/" at the end, repo will contain registry twice if repo contains registry url as well. Same case with L271. 
This needs to be fixed to ensure backward compatibility.

Reply:
Fixed by trimming off any trailing slash from registry in buildRepo()